### PR TITLE
add tests for public user project enable, sync, pause and resume

### PR DESCRIPTION
**Pull request by niranjan92**
https://github.com/Shippable/bvt-v2/issues/10

### CODE REVIEW

File: tests/core/GHC-ORG-PRI-UNA.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            projectId = project.id;
+            return done();
+          }
+        );
+      }
+    );
+
+    it('1. Cannot Enable a private project',
+      function (done) {
+        logger.info(testSuite, 'enabling project with id:', projectId);
```
Comment (shrivara) [inline tests/core/GHC-ORG-PRI-UNA.js]: Lets remove this.

File: tests/core/GHC-ORG-PRI-UNA.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        var json = {type: 'ci'};
+        global.pubAdapter.enableProjectById(projectId, json,
+          function (err) {
+            assert(err, 'public user should not be able to enable private ' +
```
Comment (shrivara) [inline tests/core/GHC-ORG-PRI-UNA.js]: Lets test for the exact result that we are expecting here. 401 unauthorized etc.

File: tests/core/GHC-ORG-PRI-UNA.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      function (done) {
+        // enable a project first
+        var json = {type: 'ci'};
+        global.suAdapter.enableProjectById(projectId, json,
+          function (err) {
+            assert(err, 'should be able to enable the project');
+
+            global.pubAdapter.syncProjectById(projectId,
+              function (err) {
+                assert(err, 'public user should not be able to sync a project');
```
Comment (shrivara) [inline tests/core/GHC-ORG-PRI-UNA.js]: Lets test for the exact result that we are expecting here. 401 unauthorized etc.

File: tests/core/GHC-ORG-PRI-UNA.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      }
+    );
+
+    it('3. Cannot pause a private project',
+      function (done) {
+        var json = {propertyBag: {isPaused: true}};
+        assert.isNotNull(projectId, 'Should have an enable project');
+        global.pubAdapter.putProjectById(projectId, json,
+          function (err) {
+            assert(err, 'public user cannot pause a private project');
```
Comment (shrivara) [inline tests/core/GHC-ORG-PRI-UNA.js]: Lets test for the exact result that we are expecting here. 401 unauthorized etc.

File: tests/core/GHC-ORG-PRI-UNA.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        var json = {propertyBag: {isPaused: true}};
+        global.suAdapter.putProjectById(projectId, json,
+          function (err) {
+            assert(!err, 'suAdapter unable be able to pause project');
+            json = {propertyBag: {isPaused: false}};
+
+            var json = {propertyBag: {isPaused: false}};
+            global.pubAdapter.putProjectById(projectId, json,
+              function (err) {
+                assert(err, 'public user should not be able to resume a ' +
```
Comment (shrivara) [inline tests/core/GHC-ORG-PRI-UNA.js]: Lets test for the exact result that we are expecting here. 401 unauthorized etc.

File: tests/core/GHC-ORG-PRI-UNA.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            assert(err, 'public user cannot pause a private project');
+            return done();
+          }
+        );
+      }
+    );
+
+    it('4. Cannot resume a private project',
+      function (done) {
+                // first enable using su adapter
```
Comment (shrivara) [inline tests/core/GHC-ORG-PRI-UNA.js]: Weird indent.

File: tests/core/GHC-ORG-PRI-UNA.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        var json = {propertyBag: {isPaused: true}};
+        global.suAdapter.putProjectById(projectId, json,
+          function (err) {
+            assert(!err, 'suAdapter unable be able to pause project');
+            json = {propertyBag: {isPaused: false}};
+
+            var json = {propertyBag: {isPaused: false}};
```
Comment (shrivara) [inline tests/core/GHC-ORG-PRI-UNA.js]: Multiple definitions of json. Did eslint not catch this? I think we should add eslint config in a shippable.yml now.

File: tests/core/GHC-ORG-PRI-UNA.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+      }
+    );
+
+    it('2. Cannot Synchonize a private project',
+      function (done) {
+        // enable a project first
+        var json = {type: 'ci'};
+        global.suAdapter.enableProjectById(projectId, json,
+          function (err) {
+            assert(err, 'should be able to enable the project');
```
Comment (shrivara) [inline tests/core/GHC-ORG-PRI-UNA.js]: Is this assertion correct?

File: tests/core/GHC-ORG-PRI-UNA.js (hunk 7 above)
Comment (niranjan92) [inline tests/core/GHC-ORG-PRI-UNA.js]: will add a rule to check shadow variables. it didn't give an error as previous definition was in a different scope.

**Pull request merged**